File: pset1/mario/mario.c
#include <stdio.h>
#include <cs50.h>


//the user doesn’t, in fact, input a positive integer between 1 and 8, inclusive, when prompted, the program should re-prompt the user until they cooperate:

int main(void)
{
    int user_input = 0;
    // get user_input
    do
    {
        user_input = get_int("Height: ");
        
    } 
    while (user_input < 1 || user_input > 8);
    // store the required space to form the hiearachy shape
    int required_hiearachy_space = user_input ; 
    
    for (int height = 1; height < user_input + 1; height++)
    {
        // adding space to the # to make it look like a hiearachy
        for (int add_space = 1; add_space < required_hiearachy_space; add_space++)
        {
            printf(" ");
        }
        // adding blocks to the hiearachy
        for (int width = 1; width <= 2 * height; width++)
        {
            printf("#");
            // if 
            if (width == height)
            {
                printf("  ");
            }
        }
        printf("\n");
        // Decrement the spaces needed for the hiearachy
        required_hiearachy_space--;
    }
    
}

File: pset6/mario/more/mario.py
from cs50 import get_int
from cs50 import get_string
'''
Height: 4
   #  #
  ##  ##
 ###  ###
####  ####
1-prompt the user with get_int for the half-pyramid’s height, a positive integer between 1 and 8
'''


def main():
    # get height
    height = get_input()
    # building block vertically
    for height_y in range(1, height + 1):
        # add space
        for space in range(height - height_y):
            add_space()
        # building block horizintally
        for width_x in range(1, (height_y * 2) + 1):
            add_blocks(width_x, height_y)
        add_new_line()
        
        
def get_input():
    # verify input is >! and input < 8
    while True:
        height = get_height()
        if not(height < 1 or height > 8):
            break
    return height


def get_height():
    return get_int("Height:")


def add_space():
    print(" ", end="")
    
    
def add_blocks(width, height):
    print("#",  end='')
    if(width == height):
        print("  ", end='')
        
        
def add_new_line():
    print("")
    
    
if __name__ == "__main__":
    main()
